docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,172 @@
+in the real cluster as needed.
+
+```shell
+clusterctl alpha topology plan -f input.yaml -o output/
+```
+
+> Note: This command can be used with or without a management cluster. In case the command is used without a management cluster the input should have
+> all the objects needed to perform the dry run. In case the command is run with a management cluster the dry run operation will look for objects 
+> in the management cluster as needed.
+
+## Dry run
+
+The operation is composed of the following steps:
+* Set the namespace on objects in the input with missing namespace.
+* Run the Defaulting and Validation webhooks on the Cluster and ClusterClass objects in the input.
+* Dry run the topology reconciler on the target cluster.
+* Capture all changes observed during reconciliation.
+
+<aside class="note">
+
+<h1>Set missing namespace</h1>
+
+The namespace used for objects with missing namespace follows this precedence order:
+* Value passed using `--namespace`
+* Current namespace defined in kubeconfig
+* Value `default`
+
+</aside>
+
+<aside class="note warning">
+
+<h1>Defaulting and Validation of objects</h1>
+
+Defaulting and validation is only performed on the Cluster and ClusterClass objects in the input. Objects of all other kinds in the input should 
+be fully valid and any required default values should be set. Missing values will not be defaulted.
+
+> **Note on Validation**: If the command is used with a real management cluster, objects from the management cluster will be used to perform 
+> `ValidateUpdate` appropriately.
+
+</aside>
+
+## Input
+
+### `--file`, `-f` (REQUIRED)
+ 
+The input file with the target changes.
+The objects in the input should follow these rules:
+* All the objects in the input should belong to the same namespace.
+* Should not have multiple Clusters.
+* Should not have multiple ClusterClasses.
+
+<aside class="note warning">
+
+<h1>Object namespaces</h1>
+
+If some of the objects have a defined namespace and some do not, the objects are considered as belonging to different namespaces. The input
+will be considered as invalid.
+
+</aside>
+
+### `--output-directory`, `-o` (REQUIRED)
+
+Information about the objects that are created and updated during dry run are written to this directory.
+
+### `--cluster`, `-c` (Optional)
+
+When multiple clusters are affected by the input, `--cluster` can be used to specify a target cluster. 
+
+If only one cluster is affected or if a Cluster is in the input it defaults as the target cluster. 
+
+### `--namespace`, `-n` (Optional)
+
+Namespace used for objects with missing namespaces in the input.
+
+## Example use cases
+ 
+### Creating a new Cluster using a new ClusterClass
+
+The command can be used to dry run creating a new Cluster along with a new ClusterClass.
+```shell
+clusterctl alpha topology plan -f new-cluster-and-class.yaml -o output/
```

Comment:
nit, up to you. Could be a good place to just show implicitly that you can use multiple `-f` args, e.g.
`-f cluster.yaml -f clusterclass.yaml`. 
I also assume usually users would store those in separate files whenever possible (as you ideally have a ~ 1:n relationship between a ClusterClass and Cluster)